src/core/api/border_agent_api.cpp

```diff
@@ -56,4 +56,11 @@ uint16_t otBorderAgentGetUdpPort(otInstance *aInstance)
     return instance.Get<MeshCoP::BorderAgent>().GetUdpPort();
 }
 
+void otBorderAgentStart(otInstance *aInstance)
```

Comment:
Why do we need this API? Why not start BorderAgent right after OT Instance construction (maybe in `MeshCoP::BorderAgent` constructor).